fastlane_core/lib/fastlane_core/tool_collector.rb

```diff
@@ -0,0 +1,72 @@
+module FastlaneCore
+  class ToolCollector
+    TOOLS = %w(fastlane fastlane_core deliver snapshot frameit pem sigh produce cert gym pilot credentials_manager spaceship scan supply watchbuild match screengrab)
+    HOST_URL = "https://fastlane-enhancer.herokuapp.com/"
+
+    attr_reader :error
+
+    def did_launch_action(name)
+      if is_official?(name)
+        launches[name] ||= 0
+        launches[name] += 1
+      end
+    end
+
+    def did_raise_error(name)
+      if is_official?(name)
+        @error = name
+      end
+    end
+
+    # Sends the used actions
+    # Example data => [:xcode_select, :deliver, :notify, :slack]
+    def did_finish
+      return if ENV["FASTLANE_OPT_OUT_USAGE"]
+      if !did_show_message? and !Helper.is_ci?
+        UI.message("Sending Crash/Success information. More information on: https://github.com/fastlane/enhancer")
+        UI.message("No personal/sensitive data is sent. Only sharing the following:")
+        UI.message(launches)
+        UI.message(@error) if @error
+        UI.message("This information is used to fix failing actions and improve integrations that are often used.")
+        UI.message("You can disable this by adding `opt_out_usage` to your Fastfile")
```

Comment:
In this case the user might not have a `Fastfile`. Depending on if we want to use this class from fastlane too, we might want to tell the user how to set the environment variable